src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberFieldSymbol.cs

```diff
@@ -571,5 +571,13 @@ internal override bool IsDefinedInSourceTree(SyntaxTree tree, TextSpan? definedW
 
             return false;
         }
+
+        internal override void AfterAddingTypeMembersChecks(ConversionsBase conversions, DiagnosticBag diagnostics)
+        {
+            var options = (CSharpParseOptions)SyntaxTree.Options;
```

Comment:
Think you can avoid the cast here with `SyntaxNode.SyntaxTree.Options`.

Reply:
The `Options` property is still a `ParseOptions` and not a `CSharpParseOptions` when I do that unfortunately. (where's the proposal for return type covariance? 😄)

Reply:
Hmm. Feel like C# specific helpers should already be available here somewhere. Can't be the first one to need to grab the options in the symbol table 😄 . Perhaps @jcouv or @AlekseyTs knows of one.